editoast/src/schema/errors.rs

```diff
@@ -4,14 +4,13 @@ use serde::{Deserialize, Serialize};
 use strum_macros::EnumVariantNames;
 
 #[derive(Serialize, Deserialize, PartialEq, Debug, Clone)]
-#[serde(deny_unknown_fields)]
```

Comment:
`#[serde(deny_unknown_fields)]` if I keep this, deserialize does not work correctly for some tests

Reply:
In this case it's the test that should be fixed.

Reply:
It's not the tests' fault, it's serde's. flatten and deny_unknown_fields are incompatible (https://serde.rs/attr-flatten.html) so `schema::InfraError` cannot be deserialized properly, even if the serialization works. Changing how these errors are represented internally has to be discussed and is likely to impact things beyond the scope of this PR. Even more so since it overlaps with https://github.com/osrd-project/osrd/issues/4580#issuecomment-1673404336. So my suggestion would be to leave the JsonValue in the response of /infra/{id}/errors.